packages/core/src/lib/wallet/wallet.types.ts

```diff
@@ -164,6 +165,28 @@ export type InjectedWallet = BaseWallet<
   InjectedWalletBehaviour
 >;
 
+// ----- Instant Link Wallet ----- //
+
+export type InstantLinkWalletMetadata = BaseWalletMetadata & {
+  contractId: string;
+  runOnStartup: boolean;
+};
+
+export type InstantLinkWalletBehaviour = BaseWalletBehaviour & {
+  networkId: string;
+  contractId: string;
+  switchAccount(id: string): Promise<void>;
+  getAccountId(): string;
+  isSignedIn: () => Promise<boolean>;
+  getAvailableBalance: () => Promise<BN>;
```

Comment:
I only added the `getContractId()` method to be a part of the `InstantLinkWallet` type since all instant wallets need them. You are right for the rest of the methods. I created a sub-type `KeypomWalletInstant` like you described. In the future if we need to extend the `InstantLinkWallet` type we can re-add methods that are neccessary.